### PR TITLE
Add orderBy keyword and tokenType to allOracles query

### PR DESCRIPTION
**Pull request by chrisli30**

### CODE REVIEW

File: src/schema/index.js [hunk 1]
```diff
@@ -59,7 +60,7 @@ type syncInfo {
 
 type Query {
   allTopics(filter: TopicFilter, skip: Int, first: Int, orderBy:String): [Topic]!
-  allOracles(filter: OracleFilter, skip: Int, first: Int, orderBy:String): [Oracle]!
+  allOracles(filter: OracleFilter, skip: Int, first: Int, orderBy: [Order]): [Oracle]!
```
Comment (chrisli30) [inline src/schema/index.js]: If this is approved I am going to apply it to all orderBy

File: src/schema/resolvers.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
-const pubsub = require('../pubsub');
```
Comment (chrisli30) [inline src/schema/resolvers.js]: This file is not used. Should I removed it?

File: src/schema/resolvers.js [hunk 3]
```diff
@@ -122,21 +129,32 @@ module.exports = {
...
-      return await cursor.toArray();
+
+      if (!_.isEmpty(orderBy)) {
+        options.sort = [];
+
+        _.each(orderBy, (order) => {
+          options.sort.push([order.field, (order.direction === "ASC" ? 'asc' : 'desc')]);
+        });
+      }
+
```
Comment (chrisli30) [inline src/schema/resolvers.js]: limit, skip and orderBy are general keywords. If this is approved I will create a general method and include it in all queries.

File: src/schema/index.js (hunk 1 above)
Comment (dwalintukan) [reply src/schema/index.js]: Makes sense to have reusable generic types for this 👍

File: src/schema/resolvers.js (hunk 3 above)
Comment (frankobe) [reply src/schema/resolvers.js]: @chrisli30 we can hv default value for limit, like 50 or 100

File: src/schema/resolvers.js [hunk 4]
```diff
@@ -122,21 +129,32 @@ module.exports = {
...
-        cursor.skip(skip);
+        options.skip(skip);
       }
-      return await cursor.toArray();
+
+      if (!_.isEmpty(orderBy)) {
+        options.sort = [];
+
+        _.each(orderBy, (order) => {
+          options.sort.push([order.field, (order.direction === "ASC" ? 'asc' : 'desc')]);
```
Comment (frankobe) [inline src/schema/resolvers.js]: I notice direction can be null, so ensure the default order u want is `desc`. The mongo default is `asc`

**Comment by frankobe** [comment]
@chrisli30 in general the direction looks good. Can u mk changes to all needed place
Also I remember there is before/after blockNum query needed, will it come in this pr or a separate one?

File: src/schema/resolvers.js (hunk 4 above)
Comment (chrisli30) [reply src/schema/resolvers.js]: good point

File: src/schema/resolvers.js (hunk 2 above)
Comment (chrisli30) [reply src/schema/resolvers.js]: Okay, got it

File: src/schema/resolvers.js (hunk 4 above)
Comment (chrisli30) [reply src/schema/resolvers.js]: Yes current default is 'desc'.

Comment (frankobe) [reply src/schema/resolvers.js]: @chrisli30 sg, probably add a comment here

**Comment by frankobe** [comment]
implemented in #56

**Pull request closed without merging**